alpha/apps/kaltura/lib/myPartnerUtils.class.php

```diff
@@ -1831,4 +1852,197 @@ public static function getWhiteListHost(Partner $partner)
+		categoryPeer::setUseCriteriaFilter(true);
+		return $categoriesNum;
+	}
+
+	/**
+	 * calculate the number of entries using partner and status
+	 *
+	 * @param string $partnerId
+	 * @param bool $onlyReady
+	 * @return int $entriesNum
+	 */
+	public static function getEntriesForPartner($partnerId, $onlyReady = true)
+	{
+		entryPeer::setUseCriteriaFilter(false);
+		$c = new Criteria();
+		$c->addAnd(entryPeer::PARTNER_ID, $partnerId);
+		if($onlyReady)
+			$c->addAnd(entryPeer::STATUS, entryStatus::READY);
+		$entriesNum = entryPeer::doCount($c);
+		entryPeer::setUseCriteriaFilter(true);
+		return $entriesNum;
+	}
+
+	/**
+	 * calculate the number of custome metadata objects using partner and status
+	 *
+	 * @param string $partnerId
+	 * @param bool $onlyValid
+	 * @return int $metadataNum
+	 */
+	public static function getMetadataObjectsForPartner($partnerId, $onlyValid = true)
+	{
+		MetadataPeer::setUseCriteriaFilter(false);
+		$c = new Criteria();
+		$c->addAnd(metadataPeer::PARTNER_ID, $partnerId);
+		if($onlyValid)
+			$c->addAnd(MetadataPeer::STATUS, Metadata::STATUS_VALID);
+		$metadataNum = MetadataPeer::doCount($c);
+		MetadataPeer::setUseCriteriaFilter(true);
+		return $metadataNum;
+	}
+
+	/**
+	 * calculate the number of categories that the partner created after the account population
+	 * (exclude objects that were from template partner)
+	 *
+	 * @param partner $partner
+	 * @return int $partnerCategories
+	 */
+	public static function getNumOfCategoriesCreatedByPartner ($partner)
+	{
+		$partnerId = $partner->getId();
+		$categories = self::getCategoriesForPartner($partnerId, false);
+		$partnerCategories = $categories - $partner->getTemplateCategoriesNum();
+		return $partnerCategories;
+	}
+
+	/**
+	 * calculate the number of entries that the partner created after the account population
+	 * (exclude objects that were from template partner)
+	 *
+	 * @param partner $partner
+	 * @return int $partnerEntries
+	 */
+	public static function getNumOfEntriesCreatedByPartner ($partner)
+	{
+		$partnerId = $partner->getId();
+		$entries = self::getEntriesForPartner($partnerId, false);
+		$partnerEntries = $entries - $partner->getTemplateEntriesNum();
+		return $partnerEntries;
+	}
+
+	/**
+	 * calculate the number of custom metadata objects that the partner created after the account population
+	 * (exclude objects that were from template partner)
+	 *
+	 * @param partner $partner
+	 * @return int $partnerMetadata
+	 */
+	public static function getNumOfMetadataObjectsCreatedByPartner ($partner)
+	{
+		$partnerId = $partner->getId();
+		$MetadataObjects = self::getMetadataObjectsForPartner($partnerId, false);
+		$partnerMetadata = $MetadataObjects - $partner->getTemplateCustomMetadataNum();
+		return $partnerMetadata;
+	}
+
+	/**
+	 * calculate the number of entries that were changed after creation
+	 *
+	 * @param partner $partner
+	 * @return int $entriesNum
+	 */
+	public static function getNumOfEntriesChangedByPartner ($partner)
```

Comment:
Use property on partner what will be updated only when updateEntry  will called.